File: versus/dicts/dicts.py
n = 3000000
k = 200000
counts = {}
for i in range(n):
    key = str(i % k)
    counts[key] = counts.get(key, 0) + 1
total = 0
for j in range(k):
    total += j * counts[str(j)]
print(total)

File: versus/dicts/dicts.cpp
#include <cstdio>
#include <unordered_map>
#include <string>

int main() {
    long n = 3000000, k = 200000;
    std::unordered_map<std::string, long> counts;
    for (long i = 0; i < n; i++) counts[std::to_string(i % k)]++;
    long total = 0;
    for (long j = 0; j < k; j++) total += j * counts[std::to_string(j)];
    printf("%ld\n", total);
    return 0;
}
